### PR TITLE
Extend Cobbler for heterogeneous hardware

### PR DESCRIPTION
**Pull request by erhudy**
This rectifies some assumptions Cobbler makes, adds support to the cluster YAML for Cobbler profiles, and makes the cluster YAML the official source of truth (replacing the venerable cluster.txt).

### CODE REVIEW

**Comment by erhudy** [comment]
Note: ipmi_username and ipmi_password being plaintext fields obviously sucks, as does the fact that it lives in the inventory, which can't be secured with Ansible Vault. I am thinking about how to approach this down the road so that we don't need to leave this stuff in plaintext.

**Comment by semenovroman** [comment]
https://www.vaultproject.io/ should be considered for secret storage down the road.

File: bootstrap/ansible_scripts/bootstrap_deployment/enroll-all-nodes-in-cobbler.yml [hunk 1]
```diff
@@ -10,22 +10,26 @@
...
-      register: cluster_txt_stat
+    - name: check that cluster.yaml is present
+      stat: path=/bcpc/deployed/cluster.yaml
+      register: cluster_yaml_stat
 
-    - name: fail if cluster.txt is not present
-      fail: msg="Please provide a cluster.txt in /bcpc/deployed on the bootstrap node"
-      when: not cluster_txt_stat.stat.exists
+    - name: check that cluster.yml is present
+      stat: path=/bcpc/deployed/cluster.yml
```
Comment (kelvk) [inline bootstrap/ansible_scripts/bootstrap_deployment/enroll-all-nodes-in-cobbler.yml]: `cluster.yaml` instead?

Comment (kelvk) [inline bootstrap/ansible_scripts/bootstrap_deployment/enroll-all-nodes-in-cobbler.yml]: Looks like we accept either format. Should we just have one?

File: bootstrap/ansible_scripts/bootstrap_deployment/enroll-all-nodes-in-cobbler.yml [hunk 2]
```diff
@@ -10,22 +10,26 @@
...
+      register: cluster_yml_stat
+
+    - name: fail if cluster YAML (.yaml or .yml) is not present
+      fail: msg="Please provide a cluster.yaml or cluster.yml in /bcpc/deployed on the bootstrap node"
+      when: not cluster_yaml_stat.stat.exists and not cluster_yml_stat.stat.exists
 
     - name: register the cluster head nodes in cluster.txt
```
Comment (kelvk) [inline bootstrap/ansible_scripts/bootstrap_deployment/enroll-all-nodes-in-cobbler.yml]: I think the yaml format is intended to be used here

**Comment by kelvk** [comment]
Will docs be updated as part of this PR?

File: bootstrap/ansible_scripts/bootstrap_deployment/enroll-all-nodes-in-cobbler.yml (hunk 2 above)
Comment (erhudy) [inline bootstrap/ansible_scripts/bootstrap_deployment/enroll-all-nodes-in-cobbler.yml]: Will update as part of docs PR.

**Comment by erhudy** [comment]
I'll add another changeset with documentation changes.

**Comment by erhudy** [comment]
Also will add another changeset that either adds a counterpart to `create_random_cluster_txt.py` to generate a similar random cluster YAML file for validation purposes or extends that script to do YAML as well. I don't intend to remove it entirely because we still need to verify that the text to YAML conversion works correctly so that we can continue to ingest cluster.txt files.

**Comment by kelvk** [comment]
`enroll_cobbler.py` should execute `cobbler sync` when changes are made.

**Comment by kelvk** [comment]
I was able to bootstrapped a fresh cluster via Ansible method successfully.

When enrolling a machine in Cobbler with non-existent profile, it fails as expected:
```
root@ansible-bcpc-bootstrap:/bcpc/deployed# /bcpc/deployed/bootstrap/ansible_scripts/scripts/enroll_cobbler.py add_host -t ansible-bcpc-vm1 -c /bcpc/deployed/cluster.yml
WARNING:root:Executing cobbler system list
WARNING:root:Executing cobbler distro list
WARNING:root:Executing cobbler profile list
WARNING:root:Executing cobbler system add --name=ansible-bcpc-vm1 --hostname=ansible-bcpc-vm1.bcpc.example.com --profile=bcpc_host1 --ip-address=10.0.100.11 --mac=08:00:27:48:6A:6F --interface=eth0
Traceback (most recent call last):
  File "/bcpc/deployed/bootstrap/ansible_scripts/scripts/enroll_cobbler.py", line 243, in <module>
    main()
  File "/bcpc/deployed/bootstrap/ansible_scripts/scripts/enroll_cobbler.py", line 232, in main
    cobb.add_host(args.target)
  File "/bcpc/deployed/bootstrap/ansible_scripts/scripts/enroll_cobbler.py", line 162, in add_host
    (cobbler_add.returncode, cobbler_add.stderr.read()))
RuntimeError: cobbler add returned 1:
```
However, could we make the messages more user-friendly?

**Pull request merged**